src/Plugin/media/Source/MediaAvPortalSourceInterface.php

```diff
@@ -9,4 +9,38 @@
 /**
  * Common interface for source plugins that use the AV Portal.
  */
-interface MediaAvPortalSourceInterface extends MediaSourceFieldConstraintsInterface {}
+interface MediaAvPortalSourceInterface extends MediaSourceFieldConstraintsInterface {
+
+  /**
+   * Gets list of supported url formats.
+   */
+  public function getSupportedUrlFormats(): array;
+
+  /**
+   * Gets list of supported url patterns.
+   */
+  public function getSupportedUrlPatterns(): array;
+
+  /**
+   * Transforms url with resource reference to a simple reference string.
+   *
+   * @param array $url
+   *   The url to transform.
+   *
+   * @return string
+   *   The transformed url.
+   */
+  public function transformUrlToReference(array $url): ?string;
```

Comment:
Why do we intake an array for URL here? Can we not just get the URL itself?

If we really need to take an array, then we need to thoroughly document the components of this expected array. Also, usually a bad idea to pass data like this in array.